sdk/highlight-run/src/sdk/util.ts

```diff
@@ -8,6 +11,31 @@ export const internalLog = (
 	void reportLog(prefix, ...msg)
 }
 
+/**
+ * Utility to help avoid logging the same message multiple times.
+ *
+ * This is helpful to prevent spamming the logs with sticky error conditions.
+ * For example if local storage cannot be written to, then each attempt to
+ * write would fail, and logging each time would spam the logs.
+ *
+ * @param logOnceId - A message with this ID will only be logged one time for the given context.
+ * @param context - The context of the message.
+ * @param level - The level of the message.
+ * @param msg - The message to log.
+ */
+export const internalLogOnce = (
```

Comment:
mind adding a changeset so we can release this functionality?

Reply:
Yeah, I will add one.